src/app/clusters/basic-information/BasicInformationCluster.cpp

```diff
@@ -45,10 +41,36 @@ using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::BasicInformation;
 using namespace chip::app::Clusters::BasicInformation::Attributes;
 using namespace chip::DeviceLayer;
-using chip::Protocols::InteractionModel::Status;
 
 namespace {
 
+// fixed size for location strings
+static constexpr size_t kFixedLocationLength = 2;
+
+constexpr DataModel::AttributeEntry kMandatoryAttributes[] = {
+    DataModelRevision::kMetadataEntry,
+    VendorName::kMetadataEntry,
+    VendorID::kMetadataEntry,
+    ProductName::kMetadataEntry,
+    ProductID::kMetadataEntry,
+    NodeLabel::kMetadataEntry,
+    Location::kMetadataEntry,
+    HardwareVersion::kMetadataEntry,
+    HardwareVersionString::kMetadataEntry,
+    SoftwareVersion::kMetadataEntry,
+    SoftwareVersionString::kMetadataEntry,
+    CapabilityMinima::kMetadataEntry,
+    SpecificationVersion::kMetadataEntry,
+    MaxPathsPerInvoke::kMetadataEntry,
+    ConfigurationVersion::kMetadataEntry,
+    // NOTE: UniqueID used to NOT be mandatory in previous spec version, so we add
+    // this as a separate condition
+    // UniqueID::kMetadataEntry,
+
+};
+
+static_assert(sizeof(bool) == 1, "I/O assumption for backwards compatibility");
```

Comment:
Note that this says nothing about bool being 8 bits or one byte; just means it's the same size as char (which is not guaranteed to by one byte by the standard).

So it's not really clear to me what this is really asserting and why....